packages/grafana-data/src/types/logs.ts

```diff
@@ -260,3 +260,37 @@ export const hasLogsContextUiSupport = (datasource: unknown): datasource is Data
 
   return withLogsSupport.getLogRowContextUi !== undefined;
 };
+
+export interface AnalyzeQueryOptions {
+  check: string;
+  attributes: KeyValue<string>;
+}
+
+/**
+ * Data sources that support query manipulation through `modifyQuery` and `analyzeQuery` in Explore.
+ * Allows for interactions such as changing the query from logs details, or displaying filter status.
+ * @internal
+ */
+export interface DataSourceWithQueryManipulationSupport<TQuery extends DataQuery, TAnalyzeQueryResult = boolean> {
+  /**
+   * Used in explore
+   */
+  modifyQuery(query: TQuery, action: QueryFixAction): TQuery;
```

Comment:
Let's do it in two steps. But, out of curiosity, who do we think could be affected by this change?

Reply:
so i spent some time thinking about this, and could not come up with anyone who would be affected 😄, so we'll be fine probably. but it will be easier for the reviewers to see that, when the step2-PR will only contain the single typescript move, i think.

Reply:
Makes sense! Thanks for double checking this.

Reply:
Btw, one possible scenario would be plugin developers using that method as a type, i.e. `DataSourceApi['modifyQuery']`. But I see that we are only deprecating it from `DataSourceApi` and not removing it 👍